docs/ha-considerations.md

```diff
@@ -317,27 +397,59 @@ metadata:
   namespace: kube-system
 spec:
   containers:
-  - command:
-    - /kube-vip
-    - start
-    - -c
-    - /vip.yaml
-    image: 'plndr/kube-vip:0.1.1'
+  - args:
+    - manager
+    env:
+    - name: vip_arp
+      value: "false"
+    - name: port
+      value: "6443"
+    - name: vip_interface
+      value: ens192
+    - name: vip_cidr
+      value: "32"
+    - name: cp_enable
+      value: "true"
+    - name: cp_namespace
+      value: kube-system
+    - name: vip_ddns
+      value: "false"
+    - name: bgp_enable
+      value: "true"
+    - name: bgp_routerid
+      value: 192.168.0.2
+    - name: bgp_as
+      value: "65000"
+    - name: bgp_peeraddress
+    - name: bgp_peerpass
+    - name: bgp_peeras
+      value: "65000"
+    - name: bgp_peers
+      value: 192.168.0.10:65000::false,192.168.0.11:65000::false
+    - name: vip_address
+      value: 192.168.0.40
+    image: ghcr.io/kube-vip/kube-vip:v0.4.0
+    imagePullPolicy: Always
     name: kube-vip
     resources: {}
     securityContext:
       capabilities:
         add:
         - NET_ADMIN
+        - NET_RAW
         - SYS_TIME
     volumeMounts:
-    - mountPath: /vip.yaml
-      name: config
+    - mountPath: /etc/kubernetes/admin.conf
+      name: kubeconfig
+  hostAliases:
+  - hostnames:
+    - kubernetes
+    ip: 127.0.0.1
   hostNetwork: true
   volumes:
   - hostPath:
-      path: /etc/kube-vip/config.yaml
-    name: config
+      path: /etc/kubernetes/admin.conf
+    name: kubeconfig
```

Comment:
For Principle of Least Privilege, we'd better not use admin config.

Reply:
This is a requirement, as this static pod is brought up as the control plane is started and needs to be able to speak with the API server as it starts.

Reply:
LGTM except this one. 
/cc @neolit123

Reply:
No worries: the `/etc/kubernetes/admin.conf` is passed so that the go-client can speak to the API server during the `kubeadm init` and perform a leaderElection and bring up the HA virtual IP address.

Reply:
mounting the admin.conf in the container grants it super power over the cluster.
also the admin.conf certificates cannot be revoked currently and cannot be gated by RBAC.

i'd say at least ghcr.io/kube-vip/kube-vip should be a distroless image to avoid distro security leaks in the image. is it distroless already? this will make it match what we have for the other component images like KCM.

in an ideal world the client that kube-vip has should be limited to only the actions it needs - control the LE for the apiserver. signing a new kubeconfig should be a few additional commands.

```
kubeadm kubeconfig user --client-name=kube-vip --org kube-vip ...
kubectl ... # add RBAC rules for kube-vip group
...
```

but i can see that this will complicate the UX around kube-vip because it means users would also have to rotate the new kubeconfig and `kubeadm certs renew` will not do it for them

Reply:
i can see us having this with admin.conf by default but adding a NOTE that if users want to limit the access of kube-vip they should sign a custom client kubeconfig and rotate it manually on expiration.

Reply:
kube-vip is not distroless it uses `SCRATCH`.